contrib/gin-gonic/gin/gintrace.go

```diff
@@ -42,11 +43,14 @@ func Middleware(service string, opts ...Option) gin.HandlerFunc {
 		// serve the request to the next middleware
 		c.Next()
 
-		span.SetTag(ext.HTTPCode, strconv.Itoa(c.Writer.Status()))
+		status := c.Writer.Status()
+		span.SetTag(ext.HTTPCode, strconv.Itoa(status))
+		if status >= 500 && status < 600 {
```

Comment:
Can we make sure this behavior is reflected throughout all HTTP server integrations?

Reply:
I can do this as a followup PR. Probably it'll be a new function in `contrib/internal/httputil/` to cover this commonly needed check of HTTP status codes.